src/pytest_mock/plugin.py

```diff
@@ -94,10 +95,16 @@ def stopall(self) -> None:
         Stop all patchers started by this fixture. Can be safely called multiple
         times.
         """
-        for p in reversed(self._patches):
+        for p, m in reversed(self._patches_and_mocks):
             p.stop()
-        self._patches[:] = []
-        self._mocks[:] = []
+        self._patches_and_mocks.clear()
+
+    def stop(self, mock: unittest.mock.MagicMock) -> None:
+        for index, (p, m) in enumerate(self._patches_and_mocks):
+            if mock is m:
+                p.stop()
+                del self._patches_and_mocks[index]
+                break
```

Comment:
Probably it is best to raise an error if the given mock is not registered by us, to avoid mistakes being silent.

Reply:
Good point